src/rdkafka_broker.c

```diff
@@ -4749,18 +4685,25 @@ static int rd_kafka_broker_thread_main(void *arg) {
                 }
         }
 
+        rd_kafka_broker_fail(rkb, LOG_DEBUG, rd_kafka_broker_destroy_error(rk),
+                             "Broker handle is terminating");
+
+        rd_dassert(rkb->rkb_state == RD_KAFKA_BROKER_STATE_DOWN);
```

Comment:
Is this necessary and correct? Let's not force this and add this later after this patch. Might create a problem if we are reaching here somehow in some other state.

Reply:
It's `rd_dassert` so will only happen when we configure it with `--enable-devel` so we detect the failure. The state DOWN is (should be always) set just before in `rd_kafka_broker_fail`